### PR TITLE
Create CI/CD pipeline for S3 and Cloudfront

### PR DESCRIPTION
**Pull request by Sotatek-TinnNguyen**
Pipeline will trigger when push or merge to branch dev then push the build to S3 bucket and validation the CloudFront's cache

### CODE REVIEW

File: .github/workflows/dev-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: klaytn-safe-react deploy workflow
+on:
+  push:
+    branches:
+      - dev
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml:5]: Why the only `dev` branch is affected to S3 distribution? Only for developement?

File: .github/workflows/dev-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+jobs:
+  deploy:
+    permissions:
+      id-token: write
+      contents: read
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v3
+
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml:17]: Please add `name` like `Checkout source code`.

File: .github/workflows/dev-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Get AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_ARN_DEV }}
+          role-session-name: SessionForKlaytnActions
+          aws-region: ${{ secrets.AWS_REGION_DEV }}
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml:23]: Please make empty line for readability below this line.

File: .github/workflows/dev-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Installing Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "14"
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml:28]: Same as the above.

File: .github/workflows/dev-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Installing dependencies
+        run: yarn install
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml]: Same as the above.

File: .github/workflows/dev-deploy.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Change homepage and get .env
+        env:
+          PARAMETER_STORE_DEV: ${{ secrets.PARAMETER_STORE_DEV }}
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml]: For what do we need parameter store?

File: .github/workflows/dev-deploy.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        run: |
+          sed -i -e "s+\/app+\/+g" package.json
+          aws ssm get-parameter --name $PARAMETER_STORE_DEV --with-decryption --output text --query Parameter.Value > .env
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml]: Same as the above.

File: .github/workflows/dev-deploy.yml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Build static files
+        run: yarn build
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml:39]: Same as the above.

File: .github/workflows/dev-deploy.yml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: "14"
+      - name: Installing dependencies
+        run: yarn install
+      - name: Change homepage and get .env
+        env:
+          PARAMETER_STORE_DEV: ${{ secrets.PARAMETER_STORE_DEV }}
+        run: |
+          sed -i -e "s+\/app+\/+g" package.json
```
Comment (odg0318) [inline .github/workflows/dev-deploy.yml]: Please show me the example of what transition will be held for this regular expression.

File: .github/workflows/dev-deploy.yml [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_ARN_DEV }}
+          role-session-name: SessionForKlaytnActions
+          aws-region: ${{ secrets.AWS_REGION_DEV }}
+      - name: Installing Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "14"
+      - name: Installing dependencies
+        run: yarn install
+      - name: Change homepage and get .env
```
Comment (praveen-kaia) [inline .github/workflows/dev-deploy.yml]: After the **yarn install,** Could you also please write the copy commands for https://github.com/klaytn/klaytn-safe-react/tree/dev/safe-deployment-assets/v1.3.0 to move into the node_modules as mentioned in the Dockerfile.

File: .github/workflows/dev-deploy.yml (hunk 6 above)
Comment (Sotatek-TinnNguyen) [reply .github/workflows/dev-deploy.yml]: We need parameter store to save env and use it to build static files

File: .github/workflows/dev-deploy.yml (hunk 7 above)
Comment (Sotatek-TinnNguyen) [reply .github/workflows/dev-deploy.yml]: For the sed: It's the conflict of docker build and static build, we need to change home path in package.json from `/app` to `/`  .
For parameter store: It's same as the above

File: .github/workflows/dev-deploy.yml (hunk 9 above)
Comment (Sotatek-TinnNguyen) [reply .github/workflows/dev-deploy.yml]: ` "homepage": "/app",` here is it

File: .github/workflows/dev-deploy.yml (hunk 10 above)
Comment (Sotatek-TinnNguyen) [reply .github/workflows/dev-deploy.yml]: Okay, I have added it.

**Review by praveen-kaia — APPROVED** [review]

File: .github/workflows/dev-deploy.yml (hunk 1 above)
Comment (Sotatek-TinnNguyen) [reply .github/workflows/dev-deploy.yml:5]: Yes, Each environment will have different bucket and will trigger in their branch

Comment (odg0318) [reply .github/workflows/dev-deploy.yml:5]: @Sotatek-TinnNguyen Please don't close this review before we finish to dicuss.

Comment (odg0318) [reply .github/workflows/dev-deploy.yml:5]: Actually, there is no `main` branch in this repository. We can discuss it in other thread.

File: .github/workflows/dev-deploy.yml (hunk 6 above)
Comment (odg0318) [reply .github/workflows/dev-deploy.yml]: What kinds of static files are stored in parameter store? Please show me the data in `.env`.

File: .github/workflows/dev-deploy.yml (hunk 9 above)
Comment (odg0318) [reply .github/workflows/dev-deploy.yml]: This code is very tricky. Why do we need this line?

File: .github/workflows/dev-deploy.yml (hunk 7 above)
Comment (odg0318) [reply .github/workflows/dev-deploy.yml]: We will use `/` instead of `/app` here.

File: .github/workflows/dev-deploy.yml (hunk 6 above)
Comment (odg0318) [reply .github/workflows/dev-deploy.yml]: @Sotatek-TinnNguyen  All values stored in `.env` are not required. `REACT_APP_GATEWAY_URL` and `REACT_APP_ENV` can be stored by GitHub Action Secrets as we discussed.

**Review by odg0318 — APPROVED** [review]

**Pull request merged**